open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/manage/vrp/VRPConsentRequestValidatorTest.java

```diff
@@ -0,0 +1,2317 @@
+    }
+
+    @DataProvider(name = "withoutCreditorAcc")
+    public Object[][] vrpPayloadWithoutCreditorAcc() {
+        return VRPDataProviders.DataProviders.METADATA_WITHOUT_CREDITOR_ACC;
+    }
+
+    @DataProvider(name = "withoutControlParameters")
+    public Object[][] vrpPayloadWithoutControlParameterKey() {
+        return VRPDataProviders.DataProviders.METADATA_WITHOUT_CONTROL_PARAMETER;
+    }
+
+    @DataProvider(name = "withoutAmountKey")
+    public Object[][] vrpPayloadWithoutAmountKey() {
+        return VRPDataProviders.DataProviders.METADATA_WITHOUT_AMOUNT;
+    }
+
+
+    @Test
+    public void testVrpPayload() {
+        JSONObject response = VRPConsentRequestValidator.validateVRPPayload("payload");
+        Assert.assertFalse((boolean) response.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+    @Test
+    public void testVrpEmptyPayload() {
+        JSONObject response = VRPConsentRequestValidator.validateVRPPayload("");
+        Assert.assertFalse((boolean) response.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+    @Test(dataProvider = "vrpDataProvider", priority = 1)
+    public void testVrpInitiation(Object payload) {
+        JSONObject response = VRPConsentRequestValidator.validateVRPPayload(JSONValue.
+                parse((String) payload));
+        Assert.assertFalse((boolean) response.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+
+    @Test(dataProvider = "vrpDataProvider", priority = 1)
+    public void testVrpControlParameters(Object payload) {
+        JSONObject response = VRPConsentRequestValidator.validateControlParameters((JSONObject) JSONValue.
+                parse((String) payload));
+        Assert.assertFalse((boolean) response.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+    @Test(dataProvider = "vrpEmptyDataProvider", priority = 1)
+    public void testVrpEmptyData(Object payload) {
+        JSONObject response = VRPConsentRequestValidator.validateVRPPayload(JSONValue.
+                parse((String) payload));
+        Assert.assertFalse((boolean) response.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+    @Test(dataProvider = "vrpDataIsJsonObject", priority = 1)
+    public void testVrpDataIsJsonObject(Object payload) {
+        JSONObject response = VRPConsentRequestValidator.validateVRPPayload(JSONValue.
+                parse((String) payload));
+        Assert.assertFalse((boolean) response.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+    @Test
+    public void testVrpInitiationPayloadWithoutControlParameters() {
+
+        String initiationPayloads = VRPDataProviders.DataProviders.METADATA_VRP_WITHOUT_CONTROL_PARAMETERS;
+        JSONObject result = VRPConsentRequestValidator.validateVRPPayload(JSONValue.
+                parse(initiationPayloads));
+
+        Assert.assertFalse((boolean) result.get(ConsentExtensionConstants.IS_VALID));
+        assertFalse(result.containsKey("ControlParameters"));
```

Comment:
This assertion is incorrect. We cannot check containsKey() for a full object. Can you please check?